test/classes/test_combinatorial_complex.py

```diff
@@ -342,3 +342,96 @@ def test_order(self):
         CC = CombinatorialComplex()
         CC.add_cell([1, 2, 3, 4, 5, 6, 7, 8], rank=1)
         assert CC.order() == 8
+
+    def test_remove_nodes(self):
+        """Test for the remove_node and remove_nodes methods."""
+        example = CombinatorialComplex()
+        example.add_cell([1, 2], rank=1)
+        example.add_cell([1, 3, 2], rank=1)
+        example.add_cell([1, 2, 4, 3], rank=2)
+        example.add_cell([2, 5], rank=1)
+        example.add_cell([2, 6, 4], rank=2)
+        example.remove_node(1)
+        assert example._complex_set.hyperedge_dict == {
+            1: {
+                frozenset({2, 5}): {"weight": 1},
+                frozenset({2}): {"weight": 1},
+                frozenset({2, 3}): {"weight": 1},
+            },
+            0: {
+                frozenset({2}): {"weight": 1},
+                frozenset({3}): {"weight": 1},
+                frozenset({4}): {"weight": 1},
+                frozenset({5}): {"weight": 1},
+                frozenset({6}): {"weight": 1},
+            },
+            2: {
+                frozenset({2, 4, 6}): {"weight": 1},
+                frozenset({2, 3, 4}): {"weight": 1},
+            },
+        }
+        with pytest.raises(KeyError):
+            example.remove_nodes([1]) == KeyError("node 1 not in CombinatorialComplex")
+        example.remove_nodes([2, 5])
```

Comment:
```suggestion
        assert str(exp_exception.value) == "node 1 not in CombinatorialComplex"
        example.remove_nodes([2, 5])
```

Reply:
Please check this line once because there might be small formatting differences which will throw an error